tools/bouncer_worker/src/lib/processMonitor.js

```diff
@@ -10,6 +10,7 @@ const ProcessMonitor = {};
 const logLabel = { label: 'PROCESSMONITOR' };
 
 const dataByPid = {};
+const dataByModel = {};
```

Comment:
it's probably better to call this `dataByRid`

Reply:
do you really need both dataByPid and dataByModel? 

I think you could just reference everything by `dataByModel`?

Reply:
I think we do, the code that executes the command only knows about it's PID and the code that sends the report only knows about the RID, so simplest to leave the PID for active process, RID for after the process is completed.

Reply:
From what I can see we don't actually use the Pid inside the processMonitor (updateMemory is called by Monitor which could've been called with the `rId`: 

![image](https://user-images.githubusercontent.com/11945337/215455902-35537a09-a850-49c5-bd04-e47751b4a767.png)

the other place that reference pid is the call to `stopMonitor`, which you can call inside with rId as it has reference to `processInformation` (which is how you construct the `dataByRid` with at `startMonitor`

![image](https://user-images.githubusercontent.com/11945337/215456066-09725c2d-27a2-47ca-9ac1-c189c58c2c78.png)


So unless I'm missing something (which I often am!), you should be able to just `dataByRid` for everything?

Reply:
yes, I had some kind of mental block on thinking it needed to know the accurate PID to monitor the mem utilisation, but as we're just checking the waterlevel of total system we don't need pid.